web/front/js-json-extra/translation.json

```
@@ -231,6 +231,35 @@
         "Three-Way operator" : false
     },
 
+    "Kotlin" : {
+        "Variable declaration" : "var s = \"a variable\"",
+        "Constant declaration" : "val s = \"a constant\"",
+        "Function declaration" : "fun <T> a_function(argument: T): T = argument",
+        "Lambda declaration" : "a_lambda = {argument -> argument}",
+        "While loop" : "while (b) {\n\t...\n}",
+        "For loop" : "for (i in 0 til n) {\n\t...\n}",
+        "Foreach loop" : "for (elem in a_list) {\n\t...\n}",
+        "Do .. while loop" : "do {\n\t...\n} while (b)",
+        "Foreach index and element loop" : "for ((i, elem) in a_list.withIndex()) {\n\t...\n}",
+        "Infinite loop" : "while (true) {\n\t...\n}",
+        "if ... else if ... else" : "if (b1) {\n\t...\n}\nelse if (b2) {\n\t...\n}\nelse {\n\t...\n}",
+        "Ternary operator" : "if (b) _true else _false",
+        "Try ... Catch" : "try {\n\t...\n} catch (e: Exception) {\n\t...\n}",
+        "Switch" : "when (x) {\n\ty -> ...\nelse -> ...\n}",
+        "Type of the %" : "Truncated",
+        "Top level constructor" : "fun main() {\n\t...\n}",
+        "Get a line of STDIN" : "readln()",
+        "Get all STDIN" : "generateSequence(::readLine)",
+        "Get a char STDIN" : false,
+        "Print without a new line at the end" : "print(s)",
+        "Print with a new line at the end" : "println(s)",
+        "Implicit conversion of bool to int" : false,
+        "Implicit conversion of char to int" : false,
+        "Format a string" : "java.lang.String.format(\"%d\", n)",
```

Kotlin has its own format function (it's basically just a wrapper of java's format function but whatever)

```suggestion
        "Format a string" : "\"%d\".format(n)",
```